zephyr/include/rtos/interrupt.h

```diff
@@ -48,9 +49,7 @@ static inline void interrupt_unregister(uint32_t irq, const void *arg)
 
 static inline int interrupt_get_irq(unsigned int irq, const char *cascade)
 {
-#if defined(CONFIG_LIBRARY) || defined(CONFIG_ACE) || defined(CONFIG_CAVS) || \
-	defined(CONFIG_ZEPHYR_POSIX) || (defined(CONFIG_IMX) && !defined(CONFIG_IMX8M)) || \
-	defined(CONFIG_AMD)
+#if defined(CONFIG_LIBRARY) || defined(__ZEPHYR__) || defined(CONFIG_ARM64)
```

Comment:
I don't think this is correct. I think IMX platforms have at least had a quirk here for irq support (note: "(defined(CONFIG_IMX) && !defined(CONFIG_IMX8M))"). I do agree this is getting way too complex, so at least for Intel platforms, "__ZEPHYR__" is sufficient as we've moved all platforms to Zephyr. I think for NXP, imx8m is a special case as it uses XTOS drivers and this is going away soon -> see https://github.com/thesofproject/sof/pull/9608 ).

Reply:
Fixed.  Indeed unless there was a special config I can't see, the whole expression was just a special case for IMX8M.  (Worth pointing out that that call is vestigial anyway, there's no point in tracking IRQ numbers in SOF code in a Zephyr build, the drivers have them already)